packages/react-native-web/src/modules/ImageLoader/index.js

```diff
@@ -111,37 +120,78 @@ const ImageLoader = {
       if (typeof failure === 'function') {
         failure();
       }
-      ImageLoader.abort(requestId);
+      ImageLoader.release(requestId);
       clearInterval(interval);
     }
   },
-  has(uri: string): boolean {
+  has(uri: ?string): boolean {
     return ImageUriCache.has(uri);
   },
-  load(uri: string, onLoad: Function, onError: Function): number {
+  load(
+    source: ImageSource,
+    onLoad: (ImageResult) => void,
+    onError: Function
+  ): number {
     id += 1;
     const image = new window.Image();
-    image.onerror = onError;
-    image.onload = (e) => {
+
+    const handleLoad = () => {
       // avoid blocking the main thread
-      const onDecode = () => onLoad({ nativeEvent: e });
-      if (typeof image.decode === 'function') {
-        // Safari currently throws exceptions when decoding svgs.
-        // We want to catch that error and allow the load handler
-        // to be forwarded to the onLoad handler in this case
-        image.decode().then(onDecode, onDecode);
-      } else {
-        setTimeout(onDecode, 0);
-      }
+      const onDecode = () => {
+        ImageUriCache.add(source.uri);
```

Comment:
I've traced this being removed (by mistake) when the Image component was refactored from class based to function based component 

Since then `ImageUriCache.add/remove` was no longer used
It's not a terrible bug, but it should be causing a delay or an empty image, sometime for the initial render of the Image component 

When an image is already rendered and we use it again, it should appear the very first render, but if nothing calls `ImageUriCache.add` it sometime won't 

When there's default source and source, it might happen that the default source is shown briefly instead of the `source`